app/controllers/concerns/verify_profile_concern.rb

```diff
@@ -0,0 +1,32 @@
+module VerifyProfileConcern
+  private
+
+  def profile_or_verify_profile_path
+    public_send "#{profile_or_verify_profile_route}_path"
+  end
+
+  def profile_or_verify_profile_url
+    public_send "#{profile_or_verify_profile_route}_url"
+  end
+
+  def profile_or_verify_profile_route
+    return 'profile' if idv_context? || profile_context?
+    return 'profile' unless current_user.decorate.pending_profile_requires_verification?
+    verify_profile_route
+  end
+
+  def verify_profile_route
+    decorated_user = current_user.decorate
+    if decorated_user.needs_profile_phone_verification?
+      flash[:notice] = t('profile.index.verification.instructions')
+      return 'verify_profile_phone'
+    end
+    return 'verify_profile' if decorated_user.needs_profile_usps_verification?
+    'profile'
```

Comment:
I would rather have an exception than drop it entirely. Otherwise, it's like a series of `if/elsif` with no final `else` -- which leaves open the possibility that someone changes the logic elsewhere and we end up breaking this routing logic.